agora-testlib/Test/Specification.hs

```diff
@@ -253,3 +255,36 @@ effectFailsWith ::
   ScriptContext ->
   SpecificationTree
 effectFailsWith tag eff datum = validatorFailsWith tag eff datum ()
+
+testValidator ::
```

Comment:
This function seems kind of useless. Having `validatorSucceedsWith` and `ValidatorFailsWith` feels more intuitive.

Reply:
Hmmm, they're currently used in *all* the proposal samples.

Reply:
`mkTestTree` simple wraps around it so that I can clearly show that if a sample is valid for the validator/policy or not in the `Spec.Proposal` module.

Reply:
Would be happy to see a better solution though, cause I feel like using `Bool` is a little bit confusing in some places.